terraform/shared/shared.tf.json.template.py

```diff
@@ -59,6 +65,44 @@
                         }
                     ]
                 })
+            },
+            'aws_config': {
+                # https://docs.aws.amazon.com/config/latest/developerguide/s3-bucket-policy.html
+                'bucket': '${aws_s3_bucket.aws_config.id}',
+                'policy': json.dumps({
+                    'Version': '2012-10-17',
+                    'Statement': [
+                        {
+                            'Sid': sid,
+                            'Effect': 'Allow',
+                            'Principal': {
+                                'Service': 'config.amazonaws.com'
+                            },
+                            'Action': action,
+                            'Resource': (
+                                '${aws_s3_bucket.aws_config.arn}'
+                                f'/*/AWSLogs/{config.aws_account_id}/Config/*'
+                                if action == 's3:PutObject' else
```

Comment:
I think this would be less complicated if there were no conditional and two statement literals. One for buckets and one for objects. It's gonna be more code, some duplication but will be easier to read.